modules/app-config/types.ts

```diff
@@ -0,0 +1,14 @@
+export interface FeatureFlagType {
+	configKey: AppConfigKey
+	title: string
+	group: AppConfigGroupKey
+	active: boolean
+}
+
+export enum AppConfigKey {
+	TestDataKey = 'TestData',
+}
+
+export enum AppConfigGroupKey {
+	test = 'test',
+}
```

Comment:
Couple of things:

- Can we get away with _not_ using the Key suffix in the type name? (e.g. can we use the `keyof` operator in usage sites instead?)
- `group` and `configKey` seem related, but they're in unrelated types and it seems like they would require parallel maintenance. It's not entirely clear to me how I should go about adding a new key based on this interface alone.
  - Can we add a type that contains both the `key` and the `group` and use that as the means to identify the flag? It's not entirely clear to me that that's _better_, but it's an alternative.

Reply:
Let me know if you still see room for improvement with `keyof` here, otherwise I've taken your ideas and implemented the `group` to be included in the `configKey`.